modules/seqexec/engine/src/main/scala/seqexec/engine/Event.scala

```diff
@@ -10,51 +10,53 @@ import gem.Observation
 import seqexec.model.ClientId
 import seqexec.model.UserDetails
 import seqexec.model.StepId
+import seqexec.engine.SystemEvent._
+import seqexec.engine.UserEvent._
 
 /**
   * Anything that can go through the Event Queue.
   */
-sealed abstract class Event[+D<:Engine.Types]
-final case class EventUser[D<:Engine.Types](ue: UserEvent[D]) extends Event[D]
-final case class EventSystem(se: SystemEvent) extends Event[Nothing]
+sealed trait Event[+F[_], +D <: Engine.Types] extends Product with Serializable
 
 object Event {
+  final case class EventUser[D <: Engine.Types](ue: UserEvent[D]) extends Event[Nothing, D]
```

Comment:
Likely half of the changes are due to putting ADT instances in the companion objects